src/SonarScanner.MSBuild/BootstrapperClass.cs

```diff
@@ -34,6 +40,13 @@ public class BootstrapperClass
         public const int ErrorCode = 1;
         public const int SuccessCode = 0;
 
+#if NETFRAMEWORK
+
+        private const string WarningMessage = "From [Date], new versions of this scanner will no longer support .NET framework runtime environments less than .NET Framework 4.6.2." +
```

Comment:
IMO you can skip having this as a field and move it together as `const string` - either way the compiler does the optimization and you'll keep all NETFRAMEWORK logic together.